proto/kvrpcpb.proto

```diff
@@ -561,11 +561,19 @@ message SplitRegionResponse {
 // Sent from TiFlash to a TiKV node.
 message ReadIndexRequest{
     Context context = 1;
+
+    // TiKV checks the given range if there is any unapplied lock
+    // blocking the read request.
+    uint64 start_ts = 2;
+    repeated KeyRange ranges = 3;
```

Comment:
What if `ranges` is empty ? Does it mean full range of this region ?

Reply:
If `ranges` is empty, no range will be checked. The start and end bound of the region should be passed if ithe full range needs to be checked.